src/lib.rs

```diff
@@ -1,7 +1,231 @@
+}
+
+pub fn stringify_request(name: &str, args: &[Value]) -> Result<String> {
+    let mut buf = Vec::new();
+    let mut writer = Writer::new(&mut buf);
+
+    writer
+        .write(br#"<?xml version="1.0" encoding="utf-8"?>"#)
+        .map_err(error::EncodingError::from)?;
+
+    writer.write_start_tag(b"methodCall")?;
+    writer.write_tag(b"methodName", name)?;
+
+    writer.write_start_tag(b"params")?;
+    for value in args {
+        writer.write_start_tag(b"param")?;
+
+        writer
+            .write(value.stringify()?.as_ref())
+            .map_err(error::EncodingError::from)?;
+
+        writer.write_end_tag(b"param")?;
+    }
+    writer.write_end_tag(b"params")?;
+    writer.write_end_tag(b"methodCall")?;
+
+    Ok(String::from_utf8(buf).map_err(error::EncodingError::from)?)
+}
+
 #[cfg(test)]
 mod tests {
+    use super::*;
+
+    #[test]
+    fn test_stringify_request() {
+        assert_eq!(
+            stringify_request("hello world", &[]).unwrap(),
+            r#"<?xml version="1.0" encoding="utf-8"?><methodCall><methodName>hello world</methodName><params></params></methodCall>"#.to_owned()
+        )
+    }
+
+    /// A 32-bit signed integer (`<i4>` or `<int>`).
+    #[test]
+    fn parse_int_values() {
+        assert_eq!(
+            parse_value("<value><i4>42</i4></value>").unwrap().as_i32(),
+            Some(42)
+        );
+
+        assert_eq!(
+            parse_value("<value><int>42</int></value>")
+                .unwrap()
+                .as_i32(),
+            Some(42)
+        );
+    }
+
+    /// A 64-bit signed integer (`<i8>`).
+    #[test]
+    fn parse_long_values() {
+        assert_eq!(
+            parse_value("<value><i8>42</i8></value>").unwrap().as_i64(),
+            Some(42)
+        );
+    }
+
+    /// A boolean value (`<boolean>`, 0 == `false`, 1 == `true`).
+    #[test]
+    fn parse_boolean_values() {
+        assert_eq!(
+            parse_value("<value><boolean>1</boolean></value>")
+                .unwrap()
+                .as_bool(),
+            Some(true)
+        );
+        assert_eq!(
+            parse_value("<value><boolean>0</boolean></value>")
+                .unwrap()
+                .as_bool(),
+            Some(false)
+        );
+    }
+
+    /// A string (`<string>`). Note that these can also appear as a raw
+    /// value tag as well.
+    #[test]
+    fn parse_string_values() {
+        assert_eq!(
+            parse_value("<value>hello</value>").unwrap().as_str(),
+            Some("hello")
+        );
+        assert_eq!(
+            parse_value("<value><string>world</string></value>")
+                .unwrap()
+                .as_str(),
+            Some("world")
+        );
+        assert_eq!(parse_value("<value />").unwrap().as_str(), Some(""));
+    }
+
+    /// A double-precision IEEE 754 floating point number (`<double>`).
+    #[test]
+    fn parse_double_values() {
+        assert_eq!(
+            parse_value("<value><double>1</double></value>")
```

Comment:
might as well give at least one of these a decimal to ensure you're parsing decimals correctly.  Might be good to make one negative too.

Reply:
Good call. Added.